includes/blocks/course-theme/class-lesson-actions.php

```diff
@@ -0,0 +1,153 @@
+	/**
+	 * Lesson_Actions constructor.
+	 */
+	public function __construct() {
+		Sensei_Blocks::register_sensei_block(
+			'sensei-lms/course-theme-lesson-actions',
+			[
+				'render_callback' => [ $this, 'render' ],
+			]
+		);
+	}
+
+	/**
+	 * Renders complete lesson button.
+	 *
+	 * @param int  $lesson_id                   Lesson ID.
+	 * @param bool $has_incomplete_prerequisite Whether it has incomplete prerequisite.
+	 *
+	 * @return string The complete lesson button.
+	 */
+	private function render_complete_lesson( $lesson_id, $has_incomplete_prerequisite ) {
+		if (
+			// Return empty if the lesson requires a quiz pass.
+			Sensei()->lesson->lesson_has_quiz_with_questions_and_pass_required( $lesson_id )
+		) {
+			return '';
+		}
+
+		// The button is disabled if the lesson requires a pre-requisite.
+		$disabled = $has_incomplete_prerequisite ? 'disabled' : '';
+
+		// The button is a secondary CTA if there is a quiz but not required to take/pass it.
+		$button_style = 'is-primary';
+		if ( Sensei_Lesson::lesson_quiz_has_questions( $lesson_id ) ) {
+			$button_style = 'is-secondary';
+		}
+
+		// Render "Mark Complete" button.
+		$nonce     = wp_nonce_field( 'woothemes_sensei_complete_lesson_noonce', 'woothemes_sensei_complete_lesson_noonce', false, false );
+		$permalink = esc_url( get_permalink() );
+		$text      = esc_html( __( 'Complete lesson', 'sensei-lms' ) );
+
+		return ( '
+			<form class="sensei-course-theme-lesson-actions__complete-lesson-form" method="POST" action="' . $permalink . '">
+				' . $nonce . '
+				<input type="hidden" name="quiz_action" value="lesson-complete" />
+				<button type="submit" class="sensei-course-theme__button ' . $button_style . '" ' . $disabled . '>
+					' . $text . '
+				</button>
+			</form>
+		' );
+	}
+
+	/**
+	 * Renders take quiz button.
+	 *
+	 * @param int  $lesson_id                   Lesson ID.
+	 * @param int  $user_id                     User ID.
+	 * @param bool $has_incomplete_prerequisite Whether it has incomplete prerequisite.
+	 *
+	 * @return string The take quiz button.
+	 */
+	private function render_take_quiz( $lesson_id, $user_id, $has_incomplete_prerequisite ) {
+		$quiz_permalink = Sensei()->lesson->get_quiz_permalink( $lesson_id );
+
+		if ( empty( $quiz_permalink ) ) {
+			return '';
+		}
+
+		if ( Sensei()->lesson->is_quiz_submitted( $lesson_id, $user_id ) ) {
+			return '';
+		}
+
+		$disabled       = $has_incomplete_prerequisite ? 'aria-disabled="true"' : '';
+		$quiz_permalink = esc_url( $quiz_permalink );
+		$text           = esc_html__( 'Take quiz', 'sensei-lms' );
+
+		return '<a href="' . $quiz_permalink . '" class="sensei-course-theme__button is-primary" ' . $disabled . '>'
+			. $text .
+		'</a>';
+	}
+
+	/**
+	 * Renders the block.
+	 *
+	 * @param array $attributes The block attributes.
+	 *
+	 * @access private
+	 *
+	 * @return string The block HTML.
+	 */
+	public function render( array $attributes = [] ) : string {
+		$lesson_id = get_the_ID();
```

Comment:
Let's start using `Sensei_Utils::get_current_lesson()` in these blocks, since there will be cases where we are on completed quiz page, which uses the normal layout, and still need to show some of these actions:

<img width="954" alt="image" src="https://user-images.githubusercontent.com/176949/144325184-bc846a27-2537-471f-91b1-f5ab18e2d308.png">

Reply:
Updated here: https://github.com/Automattic/sensei/pull/4490/commits/00b07e5a5b95a80f3b5dd132147c814b257a5487